onnxruntime/test/providers/cpu/math/topk_op_test.cc

```diff
@@ -155,113 +161,170 @@ TEST(TopKOperator, InvalidKOpset9) {
+  RunTest(opset_version, 3, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+}
+
+TEST(TopKOperator, Top3ExplicitAxisLargestElements) {
+  top_3_explicit_axis(10);
+  top_3_explicit_axis(11);
 }
 
-TEST(TopKOperator, TopAllExplicitAxisOpset10) {
+static void top_all_explicit_axis(int opset_version) {
   std::vector<float> input_vals = {0.1f, 0.3f, 0.2f, 0.4f, 0.1f, 0.3f, 0.3f, 0.2f};
   std::vector<int64_t> input_dimensions = {4, 2};
   std::vector<float> expected_vals = {0.3f, 0.4f, 0.2f, 0.3f, 0.1f, 0.3f, 0.1f, 0.2f};
   std::vector<int64_t> expected_indices = {3, 1, 1, 0, 0, 2, 2, 3};
   std::vector<int64_t> expected_dimensions = {4, 2};
   int64_t axis = 0;
-  RunTest(10, 4, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+  RunTest(opset_version, 4, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
 }
 
-TEST(TopKOperator, TopAllExplicitAxis1DInputOpset10) {
+TEST(TopKOperator, TopAllExplicitAxisLargestElements) {
+  top_all_explicit_axis(10);
+  top_all_explicit_axis(11);
+}
+
+static void top_all_explicit_axis_1D_input(int opset_version) {
   std::vector<float> input_vals = {93.0f, 695.0f, 971.0f, 978.0f, 483.0f, 247.0f, 242.0f, 983.0f, 531.0f, 723.0f, 285.0f, 527.0f, 862.0f};
   std::vector<int64_t> input_dimensions = {13};
   std::vector<float> expected_vals = {983.0f, 978.0f, 971.0f, 862.0f, 723.0f, 695.0f, 531.0f, 527.0f, 483.0f, 285.0f, 247.0f, 242.0f, 93.0f};
   std::vector<int64_t> expected_indices = {7, 3, 2, 12, 9, 1, 8, 11, 4, 10, 5, 6, 0};
   std::vector<int64_t> expected_dimensions = {13};
   int64_t axis = 0;
-  RunTest(10, 13, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+  RunTest(opset_version, 13, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+}
+
+TEST(TopKOperator, TopAllExplicitAxis1DInputLargestElements) {
+  top_all_explicit_axis_1D_input(10);
+  top_all_explicit_axis_1D_input(11);
 }
 
-TEST(TopKOperator, Top1ExplicitAxisMultiDInputOpset10) {
+static void top_1_explicit_axis_MultiD_input(int opset_version) {
   std::vector<float> input_vals = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f};
   std::vector<int64_t> input_dimensions = {2, 2, 2};
   std::vector<float> expected_vals = {3, 4, 7, 8};
   std::vector<int64_t> expected_indices = {1, 1, 1, 1};
   std::vector<int64_t> expected_dimensions = {2, 1, 2};
   int64_t axis = 1;
-  RunTest(10, 1, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+  RunTest(opset_version, 1, input_vals, input_dimensions, expected_vals, expected_indices, expected_dimensions, false, axis);
+}
+
+TEST(TopKOperator, Top1ExplicitAxisMultiDInputLargestElements) {
+  top_1_explicit_axis_MultiD_input(10);
+  top_1_explicit_axis_MultiD_input(11);
 }
```

Comment:
Need to add some tests for:

1) largest == 0 
2) sorted == 0
3) both 1 and 2

Reply:
Added tests.